When creating s3 repository snapshot, error "Cannot invoke java.lang.Number.longValue() because requestCount is null" occurs

**Elasticsearch version** 7.8.1

**Plugins installed**: analysis-icu, repository-s3

**JVM version** `openjdk version "14.0.1" 2020-04-14 (docker)`

**OS version** `Linux 4.4.0-1072-aws #82-Ubuntu SMP Fri Nov 2 15:00:21 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux (docker)`

**Description of the problem including expected versus actual behavior**:
I've just upgraded to 7.8.1 and I'm trying to make a snapshot to a s3 repository
`curl -X PUT http://[host]:9200/_snapshot/s3_repository/snapshot_test_001 -H 'Content-Type: application/json' -d '{"indices": "[snapshot_index]", "include_global_state": false}`

I'm getting this error
```json
{
  "error": {
    "root_cause": [
      {
        "type": "null_pointer_exception",
        "reason": "Cannot invoke \"java.lang.Number.longValue()\" because \"requestCount\" is null"
      }
    ],
    "type": "null_pointer_exception",
    "reason": "Cannot invoke \"java.lang.Number.longValue()\" because \"requestCount\" is null"
  },
  "status": 500
}
```

**Steps to reproduce**:
1. `curl -X PUT http://[host]:9200/_snapshot/s3_repository`
```json
{
  "type": "s3",
  "settings": {
    "bucket": "some_bucket",
    "base_path": "example_base_path"
  }
}
```
2. Create AWS S3 credentials
3. Add S3 access key and secret:
`elasticsearch-keystore add -xf s3.client.default.access_key [key]`
`elasticsearch-keystore add -xf s3.client.default.secret_key [secret]`
4. Create bucket and folder on S3 so `s3://some_bucket/example_base_path` exists
5. Call PUT method to make snapshot

**Provide logs (if relevant)**:
```
{"type": "server", "timestamp": "2020-08-28T10:12:25,859Z", "level": "WARN", "component": "r.suppressed", "cluster.name": "xxx", "node.name": "yyy", "message": "path: /_snapshot/s3_repository/snapshot_1598609545, params: {repository=s3_repository, snapshot=snapshot_1598609545}", "cluster.uuid": "abc", "node.id": "xyz" , 
"stacktrace": ["org.elasticsearch.transport.RemoteTransportException: [zzz][10.x.x.x:9300][cluster:admin/snapshot/create]",
"Caused by: java.lang.NullPointerException: Cannot invoke \"java.lang.Number.longValue()\" because \"requestCount\" is null",
"at org.elasticsearch.repositories.s3.S3BlobStore$2.collectMetrics(S3BlobStore.java:90) ~[?:?]",
"at com.amazonaws.AmazonWebServiceClient.endClientExecution(AmazonWebServiceClient.java:819) ~[?:?]",
"at com.amazonaws.AmazonWebServiceClient.endClientExecution(AmazonWebServiceClient.java:799) ~[?:?]",
"at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5079) ~[?:?]",
"at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5000) ~[?:?]",
"at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4994) ~[?:?]",
"at com.amazonaws.services.s3.AmazonS3Client.listObjects(AmazonS3Client.java:895) ~[?:?]",
"at org.elasticsearch.repositories.s3.S3BlobContainer.lambda$executeListing$16(S3BlobContainer.java:298) ~[?:?]",
"at java.security.AccessController.doPrivileged(AccessController.java:312) ~[?:?]",
"at org.elasticsearch.repositories.s3.SocketAccess.doPrivileged(SocketAccess.java:42) ~[?:?]",
"at org.elasticsearch.repositories.s3.S3BlobContainer.executeListing(S3BlobContainer.java:298) ~[?:?]",
"at org.elasticsearch.repositories.s3.S3BlobContainer.listBlobsByPrefix(S3BlobContainer.java:250) ~[?:?]",
"at org.elasticsearch.repositories.blobstore.BlobStoreRepository.listBlobsToGetLatestIndexId(BlobStoreRepository.java:1609) ~[elasticsearch-7.8.1.jar:7.8.1]",
"at org.elasticsearch.repositories.blobstore.BlobStoreRepository.latestIndexBlobId(BlobStoreRepository.java:1591) ~[elasticsearch-7.8.1.jar:7.8.1]",
"at org.elasticsearch.repositories.blobstore.BlobStoreRepository.getRepositoryData(BlobStoreRepository.java:1185) ~[elasticsearch-7.8.1.jar:7.8.1]",
"at org.elasticsearch.snapshots.SnapshotsService$2.doRun(SnapshotsService.java:334) ~[elasticsearch-7.8.1.jar:7.8.1]",
"at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:695) ~[elasticsearch-7.8.1.jar:7.8.1]",
"at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.8.1.jar:7.8.1]",
"at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[?:?]",
"at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[?:?]",
"at java.lang.Thread.run(Thread.java:832) [?:?]"] }
```

--- a/plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java
+++ b/plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java
@@ -25,6 +25,8 @@ import com.amazonaws.metrics.RequestMetricCollector;
 import com.amazonaws.services.s3.model.CannedAccessControlList;
 import com.amazonaws.services.s3.model.StorageClass;
 import com.amazonaws.util.AWSRequestMetrics;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.elasticsearch.cluster.metadata.RepositoryMetadata;
 import org.elasticsearch.common.blobstore.BlobContainer;
 import org.elasticsearch.common.blobstore.BlobPath;
@@ -39,6 +41,8 @@ import java.util.Map;
 import java.util.concurrent.atomic.AtomicLong;
 
 class S3BlobStore implements BlobStore {
+
+    private static final Logger logger = LogManager.getLogger(S3BlobStore.class);
 
     private final S3Service service;
 
@@ -105,8 +109,10 @@ class S3BlobStore implements BlobStore {
     private long getRequestCount(Request<?> request) {
         Number requestCount = request.getAWSRequestMetrics().getTimingInfo()
             .getCounter(AWSRequestMetrics.Field.RequestCount.name());
-        assert requestCount != null;
-
+        if (requestCount == null) {
+            logger.warn("Expected request count to be tracked for request [{}] but found not count.", request);
+            return 0L;
+        }
         return requestCount.longValue();
     }
 

